### PR TITLE
[dart][dart-dio] Enable strong-mode and strict types

### PR DESCRIPTION
**Pull request by kuhnroyal**
The strict typing will make it easier to eventually transition to NNBD and prevents potential type errors.
Also fix some formatting things.

The same needs to be done for normal dart.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/dart-dio/api.mustache [hunk 1]
```diff
@@ -43,15 +43,15 @@ class {{classname}} {
         {{#queryParams}}
         queryParams[r'{{baseName}}'] = {{paramName}};
         {{/queryParams}}
-        queryParams.removeWhere((key, value) => value == null);
-        headerParams.removeWhere((key, value) => value == null);
+        queryParams.removeWhere((key, dynamic value) => value == null);
```
Comment (agilob) [inline modules/openapi-generator/src/main/resources/dart-dio/api.mustache:46]: if you put `dynamic` before `value` then for consistency we should have `String` before `key`

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/resources/dart-dio/api.mustache:46]: Due to `implicit-dynamic: false`, you always have to be explicit about `dynamic`. But the `String` key is inferred.

**Review by agilob — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/dart-dio/analysis_options.mustache [hunk 2]
```diff
@@ -1 +1,6 @@
-analyzer:
\ No newline at end of file
+analyzer:
+  language:
+    strict-inference: true
+    strict-raw-types: true
+  strong-mode:
+    implicit-dynamic: false
```
Comment (Grohden) [inline modules/openapi-generator/src/main/resources/dart-dio/analysis_options.mustache:6]: i see that there's no `implicit-casts:false` in this one, maybe thats the difference between mine and your usage?

**Review by Grohden — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/dart-dio/analysis_options.mustache (hunk 2 above)
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/resources/dart-dio/analysis_options.mustache:6]: You are right, I just commented on your PR.

**Comment by kuhnroyal** [comment]
I have updated this with `implicit-casts: false`, not many changes for that here. But I also added dartanalyzer to integration tests.

**Comment by wing328** [comment]
Travis CI not related to this change.

**Pull request merged**